### PR TITLE
DR2-1258 Create secret rotation lambda.

### PR DESCRIPTION
**Pull request by MancunianSam**
This has been tested without the broken Preservica calls and it's
rotating the secret so this is ready for review.

### CODE REVIEW

File: rotate-preservation-system-password/src/main/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/Lambda.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  override def handler: (RotationEvent, Config, Dependencies) => IO[Unit] = (event, config, dependencies) =>
+    for {
+      client <- IO.pure(dependencies.secretsManagerClient(event.secretId))
+      token = event.clientRequestToken
+      secretId = event.secretId
+      describeSecretResponse <- client.describeSecret()
+      versions <- versions(describeSecretResponse)
+      _ <- IO.raiseWhen(!describeSecretResponse.rotationEnabled())(new Exception(s"Secret $secretId is not enabled for rotation"))
+      _ <- IO.raiseWhen(!versions.contains(event.clientRequestToken))(new Exception(s"Secret version $token has no stage for rotation of secret $secretId."))
```
Comment (techncl) [inline rotate-preservation-system-password/src/main/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/Lambda.scala:34]: This message is a little unclear, maybe `"Secret version $token has no stage set for rotation of secret $secretId."` or `"No stage has been set for secret version $token for rotation of secret $secretId."`?

**Review by techncl — COMMENTED** [review]
Looks good! Just a minor comment.

File: rotate-preservation-system-password/src/main/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/Lambda.scala (hunk 1 above)
Comment (MancunianSam) [reply rotate-preservation-system-password/src/main/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/Lambda.scala]: I've gone with "Secret version $token has no stage set for rotation of secret $secretId." :+1:

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**